TF_Service_dotNet/TouchFree/Connections/ClientConnectionManager.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Net.WebSockets;
+
+using Ultraleap.TouchFree.Library.Configuration;
+
+namespace Ultraleap.TouchFree.Library.Connections
+{
+    public class ClientConnectionManager : IClientConnectionManager
+    {
+        private readonly ConcurrentDictionary<Guid, IClientConnection> activeConnections = new ConcurrentDictionary<Guid, IClientConnection>();
+
+        public IEnumerable<IClientConnection> ClientConnections
+        {
+            get { return activeConnections.Values; }
+        }
+
+        public event Action LostAllConnections;
+
+        public short port = 9739;
+
+        public HandPresenceEvent MissedHandPresenceEvent { get; private set; }
```

Comment:
This isn't a c# event. It's a way of recording the hands lost/found event that we propagate to clients incase of an unexpected disconnection so that on reconnection we let them know the current hand presence state. It could maybe be renamed to avoid confusion

Reply:
Ah ok - should it be private then? Seems like it shouldn't be something exposed by the manager

Reply:
It's consumed by the ClientConnection class so needs to be accessible over the IClientConnectionManager interface so can't be private

Reply:
I think the interaction between client connection and client connection manager could be improved and simplified but that would be a story in itself